File: ProgrammingFundamentals/C#/03.DataTypesAndVariables/Exercises/P10_CenturiesToNanoseconds/P10_CenturiesToNanoseconds.cs
using System;

namespace P10_CenturiesToNanoseconds
{
    class P10_CenturiesToNanoseconds
    {
        static void Main(string[] args)
        {
            int centuries = int.Parse(Console.ReadLine());
            int years = centuries * 100;
            int days = (int)(years * 365.2422);
            long hours = (long)(days * 24);
            long minutes = hours * 60L;
            decimal seconds = minutes * 60M;
            decimal milliseconds = seconds * 1000M;
            decimal microseconds = milliseconds * 1000M;
            decimal nanoseconds = microseconds * 1000M;

            Console.WriteLine($"{centuries} centuries = {years} years = {days} days = {hours} hours = {minutes} minutes = " +
                $"{seconds} seconds = {milliseconds} milliseconds = {microseconds} microseconds = {nanoseconds} nanoseconds");
        }
    }
}

File: ProgrammingFundamentals/Java/03.DataTypesAndVariables/Exercises/src/P10_CenturiesToNanoseconds.java
import java.math.BigDecimal;
import java.util.Scanner;
public class P10_CenturiesToNanoseconds {
    public static void main(String[] args) {
        Scanner console = new Scanner(System.in);

        int centuries = Integer.parseInt(console.nextLine());
        int years = centuries * 100;
        int days = (int)(years * 365.2422);
        long hours = (long)(days * 24);
        long minutes = hours * 60L;

        BigDecimal thousands = new BigDecimal(1000);
        BigDecimal seconds = new BigDecimal(minutes*60);
        BigDecimal milliseconds = seconds.multiply(thousands);
        BigDecimal microseconds = milliseconds.multiply(thousands);
        BigDecimal nanoseconds = microseconds.multiply(thousands);

        System.out.format("%d centuries = %d years = %d days = %d hours = %d minutes = " +
                          "%s seconds = %s milliseconds = %s microseconds = %s nanoseconds",
                centuries, years, days, hours, minutes, seconds, milliseconds, microseconds, nanoseconds);
    }
}
